packages/@ember/-internals/glimmer/lib/renderer/strict-resolver.ts

```diff
@@ -0,0 +1,40 @@
+import type {
+  CompileTimeResolver as VMCompileTimeResolver,
+  InternalComponentManager,
+  Nullable,
+  ResolvedComponentDefinition,
+  RuntimeResolver as VMRuntimeResolver,
+} from '@glimmer/interfaces';
+import { BUILTIN_HELPERS, BUILTIN_KEYWORD_HELPERS } from '../resolver';
+
+///////////
+
+/**
+ * Resolution for non built ins is now handled by the vm as we are using strict mode
+ */
+export class StrictResolver implements VMRuntimeResolver, VMCompileTimeResolver {
```

Comment:
why do we need a resolver at all? can we not have one? 😅 

like, why do we still need lookupHelper?

Reply:
Originally, I thought: I want Glimmer to support the "no resolver" case, and then we wouldn't need one. We'd still want it for now to avoid blocking this PR on that Glimmer change.

As I was implementing this feature, though, I realized that keywords like `array` and `hash` and builtins used by AST transformations are *not implemented* as *either* Glimmer keywords *or* AST transformations. Instead, many of them are implemented as custom helpers looked up through the resolver. This is *definitely wrong*, but it's another thing I don't want to try to fix at the same time as the rest of this PR.